### PR TITLE
Add NEXTCLOUD_UPDATE for containers started with non-default commands 

### LINKED ISSUE #491: custom "full" build
Hello,

since version 14.1 a custom buld using the full example didn't work anymore. In the "full" example the CMD was changed from "apache-forground" to "/usr/bin/supervisord" so the installation or update process in the "/entrypoint.sh" script will never be triggered.
For my opinion the must be a third if clause so "supervisord" is also a trigger in the "/entrypoint.sh" script next "apache" and "php-fpm".

my solution is: (please be critical by checking it) file: [/entrypoint.sh]
```
#!/bin/bash
....
if [[ "$1" == *apache* ]] || [[ "$1" == *supervisord* ]] || [[ "$1" == *php-fpm* ]]; then
```

best regards
Willi

### PR DESCRIPTION
**Pull request by SnowMB**
Todo:
- [x] Add environment variable
- [x] Add description to readme
- [x] Fix examples

fixes #491

### CODE REVIEW

File: .examples/README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ The Dockerfiles use the default images as base image and build on top of it.
 
 Example | Description
 ------- | -------
-[cron](https://github.com/nextcloud/docker/tree/master/.examples/dockerfiles/cron) | uses supervisor to run the cron job inside the container (so no extra container is needed).
+[cron](https://github.com/nextcloud/docker/tree/master/.examples/dockerfiles/cron) | uses supervisor to run the cron job inside the container (so no extra container is needed). This image runs `supervisord`. To enable auto install and update run images like this with an additional environment variable:  `NEXTCLOUD_UPDATE=1`. See [here](https://github.com/nextcloud/docker/blob/master/README.md#auto-configuration-via-environment-variables) for more information.
```
Comment (tilosp) [inline .examples/README.md]: It would also make sense to set it for the user in the Dockerfile

Comment (SnowMB) [reply .examples/README.md]: Can it then be overwritten by explicitly setting another value?

Comment (tilosp) [reply .examples/README.md]: I am not sure. I would only add it to the supervisor example. And if a user doesn't want it he can simply comment it out before building his custom image.

Comment (tilosp) [reply .examples/README.md]: And the supervisor example is basically useless without the updater working.

**Comment by SnowMB** [comment]
@tilosp done. Changed the wording in the readme a little bit to clarify the use of the variable.

**Comment by tilosp** [comment]
Is there some reason why this is still wip?

**Comment by tilosp** [comment]
https://github.com/nextcloud/docker/blob/3838458843b9115179791f693e7afa65d6e22eed/.travis/test-example-dockerfiles.sh#L16
this is why.

But we can simply remove the official-images test for the examples. It's not really necessary.

**Comment by SnowMB** [comment]
Ah. Does that mean this also runs into the echo test and fails because of additional outputs?

**Comment by tilosp** [comment]
> Ah. Does that mean this also runs into the echo test and fails because of additional outputs?

yes

**Comment by SnowMB** [comment]
Ok removed it. Thanks for your help.

**Pull request merged**